src/Components/Web.JS/src/Rendering/Events/EventTypes.ts

```diff
@@ -258,9 +258,10 @@ function normalizeTimeBasedValue(element: HTMLInputElement): string {
   const type = element.type;
   switch (type) {
     case 'date':
-    case 'datetime-local':
     case 'month':
       return value;
+    case 'datetime-local':
+      return value.length === 16 ? value + ':00' : value; // Convert yyyy-MM-ddTHH:mm to yyyy-MM-ddTHH:mm:00
```

Comment:
If a `step` is not specified, seconds are not included in the format, so we need to add them here for conversion to succeed. This was the cause of the problem illustrated in the code sample in #12376.

Reply:
Do you mind expanding the comment to include this?

Reply:
Also, is there a better way to detect this or handle this rather than relying on the length of the value? It seems somewhat brittle.

Reply:
If there is some web standards spec somewhere saying that the only possible formats are `yyyy-MM-ddTHH:mm` and `yyyy-MM-ddTHH:mm:ss` then the length check seems fine. But if things are more ambiguous, then yeah I agree - I guess we could do a regex looking for `^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}$` or whatever the correct syntax is.

Reply:
Sure, I could change the logic to see if the `step` property is defined and use that to determine whether we need to normalize the format like this. The case for `time` inputs works the same way so I can change that one too.

Reply:
Once again, didn't mean to ignore your comments, Steve :)

My previous idea to look at the `step` property only complicates things further, because you could still have a `step` specified and have a `:00` at the end of the input (if you programmatically set the `value` of the input, which is what some of the tests do). According to the [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time#time_value_format), the only two possible formats are indeed `HH:mm` and `HH:mm:ss` for `time` inputs. Similar idea for `datetime-local`, although MDN is a little more ambiguous about that (but I've been able to confirm similar behavior through testing).

I could leave the length check as-is, or do the regex approach if that seems better. Just to clarify, the case for `'time'` already worked using a length check (but we could change that one too).

Reply:
> Just to clarify, the case for 'time' already worked using a length check (but we could change that one too).

Oh yeah, I see that now. Well if we have precedence, leaving this as-is should be fine. I'll consider this conversation resolved.